src/scvmm/azext_scvmm/custom.py

```diff
@@ -0,0 +1,1337 @@
+
+
+def list_virtual_network(
+    cmd, client: VirtualNetworksOperations, resource_group_name=None
+):
+    if resource_group_name:
+        return client.list_by_resource_group(resource_group_name)
+    return client.list_by_subscription()
+
+
+# endregion
+
+# region VirtualMachineTemplates
+
+
+def create_vm_template(
+    cmd,
+    client: VirtualMachineTemplatesOperations,
+    resource_group_name,
+    resource_name,
+    custom_location,
+    location,
+    vmmserver=None,
+    uuid=None,
+    inventory_item=None,
+    tags=None,
+    no_wait=False,
+):
+    vm_template = VirtualMachineTemplate(
+        location=location,
+        extended_location=get_extended_location(custom_location),
+        vmm_server_id=vmmserver,
+        uuid=uuid,
+        inventory_item_id=inventory_item,
+        tags=tags,
+    )
+
+    return sdk_no_wait(
+        no_wait,
+        client.begin_create_or_update,
+        resource_group_name,
+        resource_name,
+        vm_template,
+    )
+
+
+def update_vm_template(
+    cmd,
+    client: VirtualMachineTemplatesOperations,
+    resource_group_name,
+    resource_name,
+    tags=None,
+    no_wait=False,
+):
+    vm_template_update = ResourcePatch(tags=tags)
+    return sdk_no_wait(
+        no_wait,
+        client.begin_update,
+        resource_group_name,
+        resource_name,
+        vm_template_update,
+    )
+
+
+def delete_vm_template(
+    cmd,
+    client: VirtualMachineTemplatesOperations,
+    resource_group_name,
+    resource_name,
+    force=None,
+    no_wait=False,
+):
+    return sdk_no_wait(
+        no_wait, client.begin_delete, resource_group_name, resource_name, force
+    )
+
+
+def show_vm_template(
+    cmd, client: VirtualMachineTemplatesOperations, resource_group_name, resource_name
+):
+    return client.get(resource_group_name, resource_name)
+
+
+def list_vm_template(
+    cmd, client: VirtualMachineTemplatesOperations, resource_group_name=None
+):
+    if resource_group_name:
+        return client.list_by_resource_group(resource_group_name)
+    return client.list_by_subscription()
+
+
+# endregion
+
+# region VirtualMachines
+
+# pylint: disable=too-many-locals
+def create_vm(
+    cmd,
+    client: VirtualMachinesOperations,
+    resource_group_name,
+    resource_name,
+    custom_location,
+    location,
+    vmmserver=None,
+    inventory_item=None,
+    vm_template=None,
+    cloud=None,
+    admin_password=None,
+    cpu_count=None,
+    memory_size=None,
+    dynamic_memory_enabled=None,
+    dynamic_memory_max=None,
+    dynamic_memory_min=None,
+    nics=None,
+    disks=None,
+    availability_sets=None,
+    tags=None,
+    no_wait=False,
+):
+    hardware_profile = None
+    os_profile = None
+    network_profile = None
+    storage_profile = None
+
+    if dynamic_memory_enabled is not None:
+        if dynamic_memory_enabled:
+            dynamic_memory_enabled = 'true'
+        else:
+            dynamic_memory_enabled = 'false'
```

Comment:
get_three_state_flag can support to return string value instead of boolean value. You can follow this example:
https://github.com/Azure/azure-cli/blob/564b1427d19e8862b1b8c2df8b1d00772a1cee4f/src/azure-cli/azure/cli/command_modules/network/_params.py#L862-L864

Reply:
Have removed these lines and made changes to get_three_state_flag to get string instead of boolean